tests/test_covarites_assisted_embed.py

```diff
@@ -0,0 +1,303 @@
+        self.assertEqual(casc_results['h'],results_ans['h'])
+        self.assertEqual(casc_results['wcss'],results_ans['wcss'])
+        self.assertEqual(casc_results['cascSvd'],results_ans['cascSvd'])
+        self.assertEqual(ResultARI,1)
+    def _testcasc_non_assort(self, *args, **kwargs):
+        A=([[0., 1., 0., 1., 0., 0., 0., 0., 1., 1., 1., 1., 1., 0., 0., 1.,
+        1., 1., 1., 1.],
+       [1., 0., 1., 0., 0., 1., 0., 1., 0., 0., 1., 0., 0., 0., 1., 1.,
+        1., 1., 1., 1.],
+       [0., 1., 0., 1., 0., 1., 0., 1., 0., 0., 1., 1., 1., 1., 1., 0.,
+        1., 1., 1., 1.],
+       [1., 0., 1., 0., 0., 1., 0., 0., 0., 0., 1., 1., 0., 1., 1., 1.,
+        1., 1., 1., 1.],
+       [0., 0., 0., 0., 0., 1., 0., 1., 0., 0., 1., 0., 1., 1., 1., 1.,
+        1., 1., 1., 1.],
+       [0., 1., 1., 1., 1., 0., 0., 0., 0., 0., 1., 1., 1., 1., 1., 1.,
+        1., 0., 0., 1.],
+       [0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1., 1., 1., 1., 1., 1.,
+        1., 0., 1., 0.],
+       [0., 1., 1., 0., 1., 0., 0., 0., 0., 0., 1., 1., 1., 0., 1., 1.,
+        0., 1., 1., 1.],
+       [1., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1., 1., 1., 1., 1., 1.,
+        1., 1., 0., 1.],
+       [1., 0., 0., 0., 0., 0., 0., 0., 0., 0., 1., 1., 1., 1., 1., 1.,
+        1., 1., 1., 0.],
+       [1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 0., 0., 0., 1., 0., 0.,
+        0., 0., 0., 1.],
+       [1., 0., 1., 1., 0., 1., 1., 1., 1., 1., 0., 0., 0., 0., 0., 0.,
+        1., 0., 0., 0.],
+       [1., 0., 1., 0., 1., 1., 1., 1., 1., 1., 0., 0., 0., 0., 0., 0.,
+        0., 0., 0., 0.],
+       [0., 0., 1., 1., 1., 1., 1., 0., 1., 1., 1., 0., 0., 0., 0., 0.,
+        0., 0., 0., 1.],
+       [0., 1., 1., 1., 1., 1., 1., 1., 1., 1., 0., 0., 0., 0., 0., 0.,
+        0., 0., 0., 0.],
+       [1., 1., 0., 1., 1., 1., 1., 1., 1., 1., 0., 0., 0., 0., 0., 0.,
+        0., 1., 0., 1.],
+       [1., 1., 1., 1., 1., 1., 1., 0., 1., 1., 0., 1., 0., 0., 0., 0.,
+        0., 0., 0., 1.],
+       [1., 1., 1., 1., 1., 0., 0., 1., 1., 1., 0., 0., 0., 0., 0., 1.,
+        0., 0., 1., 0.],
+       [1., 1., 1., 1., 1., 0., 1., 1., 0., 1., 0., 0., 0., 0., 0., 0.,
+        0., 1., 0., 1.],
+       [1., 1., 1., 1., 1., 1., 0., 1., 1., 0., 1., 0., 0., 1., 0., 1.,
+        1., 0., 1., 0.]])
+        covarites=([[0., 1.],
+       [0., 1.],
+       [0., 0.],
+       [0., 1.],
+       [1., 1.],
+       [1., 1.],
+       [1., 0.],
+       [1., 1.],
+       [0., 1.],
+       [0., 1.],
+       [1., 1.],
+       [1., 1.],
+       [1., 0.],
+       [1., 0.],
+       [1., 1.],
+       [1., 1.],
+       [0., 0.],
+       [1., 0.],
+       [1., 0.],
+       [1., 0.]])
+        labels=10*[0]+10*[1]
+        
+        casc=CovariateAssistedSpectralEmbed(n_components=2,assortative=False,CCA=False,check_lcc=False)
+        casc_results=casc.fit(np.array(A),covarites)
+        
+        results_ans={'cluster': ([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]),

```

Comment:
for all of these results I don't understand how you know what the right answer should be

Reply:
Because this is a situation that it will give 100% accurate clustering.(very signicificance in block probability and covarience)

Reply:
don't tell me. explain it in the code. explain what the situation is and how you know it should give perfect clustering